packages/emotion-primitives/src/createEmotion.js

```diff
@@ -0,0 +1,109 @@
+import * as React from 'react'
+import reactPrimitives from 'react-primitives'
+import PropTypes from 'prop-types'
+
+import { getStyles } from './getStyles'
+import { convertToRNStyles } from './convertToRNStyles'
+
+const isValidPrimitive = primitive =>
+  ['Text', 'View', 'Image'].indexOf(primitive) > -1
+
+const getPrimitive = primitive => {
+  if (typeof primitive === 'string' && isValidPrimitive(primitive)) {
+    return reactPrimitives[primitive]
+  } else if (typeof primitive === 'string' && !isValidPrimitive(primitive)) {
+    throw new Error(
+      `Cannot style invalid primitive ${primitive}. Expected primitive to be one of ['Text', 'View', 'Image']`
+    )
+  } else if (typeof primitive === 'function') {
+    return primitive
+  }
+}
+
+function evalStyles(context, Comp, styles, styleOverrides) {
+  // Assign static property so that the styles can be reused (like in withComponent)
+  Comp.styles = convertToRNStyles.call(context, styles)
+
+  return getStyles.call(context, Comp.styles, context.props, styleOverrides)
+}
+
+/**
+ * Creates a function that renders the styles on multiple targets with same code.
+ */
+export function createEmotionPrimitive(splitProps) {
+  /*
+   * Returns styled component
+   */
+  return function emotion(primitive, { displayName } = {}) {
+    return createStyledComponent
+
+    /**
+     * Create emotion styled component
+     */
+    function createStyledComponent() {
+      let styles = []
+
+      styles.push.apply(styles, arguments)
```

Comment:
i think this should be an equivalent of `function createStyledComponent(...styles) { ... }`

Reply:
I don't think so. This here makes sure that we receive interpolation array when using string template literal or a valid object when used object style notation. But if you've an alternate approach to this, you can alter this 😄

Reply:
not sure what do u mean, u basically spread `arguments` into `styles` array with that `push.apply`, it seems to me that `function createStyledComponent(...styles) { ... }` is 100% the same, only might be more performant (i have no data to back this up though)

could u explain how they might not be the same?

Reply:
I'll test and see if it works (doesn't break other things) without using `styles` array.